bundles/org.openhab.binding.miele/src/main/java/org/openhab/binding/miele/internal/handler/MieleBridgeHandler.java

```diff
@@ -276,50 +213,85 @@ public void run() {
+                logger.debug("The appliance with ID '{}' has been removed", cachedHomeDevice.UID);
+                for (DiscoveryListener listener : discoveryListeners) {
+                    listener.onApplianceRemoved(cachedHomeDevice);
+                }
+                ApplianceStatusListener listener = applianceStatusListeners
+                        .get(cachedHomeDevice.getApplianceIdentifier().getApplianceId());
+                if (listener != null) {
+                    listener.onApplianceRemoved();
+                }
+                cachedHomeDevicesByRemoteUid.remove(cachedHomeDevice.getRemoteUid());
+                iterator.remove();
             }
+        }
+    }
 
-            logger.debug("{} is reachable", ipAddress);
-            return true;
+    public List<HomeDevice> getHomeDevicesEmptyOnFailure() {
+        try {
+            return getHomeDevices();
+        } catch (MieleRpcException e) {
+            Throwable cause = e.getCause();
+            if (cause == null) {
+                logger.debug("An exception occurred while getting the home devices: '{}'", e.getMessage());
+            } else {
+                logger.debug("An exception occurred while getting the home devices: '{}' -> '{}", e.getMessage(),
+                        cause.getMessage());
+            }
+            return new ArrayList<>();
         }
-    };
+    }
 
-    public List<HomeDevice> getHomeDevices() {
+    private List<HomeDevice> getHomeDevices() throws MieleRpcException {
         List<HomeDevice> devices = new ArrayList<>();
 
-        if (getThing().getStatus() == ThingStatus.ONLINE) {
-            try {
-                String[] args = new String[1];
-                args[0] = "(type=SuperVision)";
-                JsonElement result = invokeRPC("HDAccess/getHomeDevices", args);
-
-                for (JsonElement obj : result.getAsJsonArray()) {
-                    HomeDevice hd = gson.fromJson(obj, HomeDevice.class);
-                    if (hd != null) {
-                        devices.add(hd);
-                    }
-                }
-            } catch (MieleRpcException e) {
-                Throwable cause = e.getCause();
-                if (cause == null) {
-                    logger.debug("An exception occurred while getting the home devices: '{}'", e.getMessage());
-                } else {
-                    logger.debug("An exception occurred while getting the home devices: '{}' -> '{}", e.getMessage(),
-                            cause.getMessage());
-                }
+        if (!isInitialized()) {
+            return devices;
+        }
+
+        String[] args = new String[1];
+        args[0] = "(type=SuperVision)";
+        JsonElement result = gatewayCommunication.invokeRPC("HDAccess/getHomeDevices", args);
+
+        for (JsonElement obj : result.getAsJsonArray()) {
+            HomeDevice hd = gson.fromJson(obj, HomeDevice.class);
```

Comment:
No need to catch exception ?

Reply:
It's a runtime exception, so technically it is not required to catch it. It should only occur either in case there's a bug in the code or if the firmware would change and produce some unexpected response. I never saw this happen (and it was never caught), and the firmware for sure will never change as Miele abandoned it. So I think it's safe to leave like is.